mrp_warehouse_calendar/models/mrp_production.py

```diff
@@ -28,3 +28,38 @@ def copy(self, default=None):
             )
             mo.date_planned_finished = date_expected
         return mo
+
+    def _get_move_finished_values(
+        self,
+        product_id,
+        product_uom_qty,
+        product_uom,
+        operation_id=False,
+        byproduct_id=False,
+        cost_share=0,
+    ):
+        res = super(MrpProduction, self)._get_move_finished_values(
+            product_id,
+            product_uom_qty,
+            product_uom,
+            operation_id,
+            byproduct_id,
+            cost_share,
+        )
+        warehouse = self.picking_type_id.warehouse_id
+        if warehouse.calendar_id:
+            if self.product_id.produce_delay:
+                self.date_planned_finished = warehouse.calendar_id.plan_days(
```

Comment:
A hook method to get the finished move values is not a good place to change the date_planned_finished of the MO. The proper place to do this fix is probably at `_onchange_date_planned_start` method.

Also, can you add a test case that fails before the fix and it is fixed with it? this will help to avoid regressions in the future

Reply:
@LoisRForgeFlow 

Checking the tests I noticed that there is already a test that checks this but it was calling an unused function.